core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -58,26 +113,260 @@ func (l *logPollerWrapper) Ready() error {
 // "public" methods of LogPollerWrapper
 func (l *logPollerWrapper) LatestRequests() ([]evmRelayTypes.OracleRequest, error) {
 	// TODO poll both active and proposed coordinators for requests and parse them
+
+	// logs, err := lp.Logs(
+	// 	end-lookbackBlocks,
+	// 	end,
+	// 	iregistry21.IKeeperRegistryMasterUpkeepPerformed{}.Topic(),
+	// 	registryAddress,
+	// 	pg.WithParentCtx(ctx),
+	// )
+	// if err != nil {
+	// 	return nil, fmt.Errorf("%w: failed to collect logs from log poller", err)
+	// }
+
+	// TODO:
+	// format logs
+
 	return nil, nil
 }
 
 func (l *logPollerWrapper) LatestResponses() ([]evmRelayTypes.OracleResponse, error) {
-	// TODO poll both active and proposed coordinators for responses and parse them
+	// TODO poll both active and proposed coordinators for requests and parse them
+
+	// logs, err := lp.Logs(
+	// 	end-lookbackBlocks,
+	// 	end,
+	// 	iregistry21.IKeeperRegistryMasterUpkeepPerformed{}.Topic(),
+	// 	registryAddress,
+	// 	pg.WithParentCtx(ctx),
+	// )
+	// if err != nil {
+	// 	return nil, fmt.Errorf("%w: failed to collect logs from log poller", err)
+	// }
+
+	// TODO:
+	// format logs
+
+	return nil, nil
+}
+
+func (l *logPollerWrapper) LatestUpdates() ([]evmRelayTypes.RouteUpdate, error) {
+	// TODO poll both active and proposed coordinators for requests and parse them
+
+	// logs, err := lp.Logs(
+	// 	end-lookbackBlocks,
+	// 	end,
+	// 	iregistry21.IKeeperRegistryMasterUpkeepPerformed{}.Topic(),
+	// 	registryAddress,
+	// 	pg.WithParentCtx(ctx),
+	// )
+	// if err != nil {
+	// 	return nil, fmt.Errorf("%w: failed to collect logs from log poller", err)
+	// }
+
+	// TODO:
+	// format logs
+
 	return nil, nil
 }
 
 // "internal" method called only by EVM relayer components
 func (l *logPollerWrapper) SubscribeToUpdates(subscriberName string, subscriber evmRelayTypes.RouteUpdateSubscriber) {
-	if l.contractVersion == 0 {
+	if l.pluginConfig.ContractVersion == 0 {
 		// in V0, immediately set contract address to Oracle contract and never update again
-		err := subscriber.UpdateRoutes(l.routerContract, l.routerContract)
-		l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "subscriberName", subscriberName, "error", err)
-	} else if l.contractVersion == 1 {
+		err := subscriber.UpdateRoutes(l.contractAddress, l.contractAddress)
+		if err != nil {
+			l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "subscriberName", subscriberName, "error", err)
+		}
+	} else if l.pluginConfig.ContractVersion == 1 {
 		l.mu.Lock()
 		defer l.mu.Unlock()
 		l.subscribers[subscriberName] = subscriber
+		err := subscriber.UpdateRoutes(l.contractAddress, l.contractAddress)
```

Comment:
We don't want to update immediately, we don't have those routes yet (SubscribeToUpdates happens before Start()). One option is to do an eager check with the client on object creation, let me think about it...